vivisect/impemu/platarch/arm.py

```diff
@@ -27,12 +27,7 @@ def __init__(self, vw, **kwargs):
 
         va = kwargs.get('va')
         if va:
-            loc = vw.getLocation(va)
-            if loc:
-                lva, lsz, lt, lflags = loc
-                self.setMemArchitecture(lflags)
-                tmode = (lflags & envi.ARCH_MASK == envi.ARCH_THUMB)
-                self.setThumbMode(tmode)
+            self._prep(va)
```

Comment:
this is the bugfix.  this causes the setup to use the long-standing (and more robust) `_prep()` code to set T-bit correctly.